pkg/container/vector/vector.go

```diff
@@ -652,6 +652,48 @@ func (v *Vector) PreExtend(rows int, mp *mpool.MPool) error {
 	return extend(v, rows, mp)
 }
 
+// PreExtendArea use to expand the mpool and area of vector
+func (v *Vector) PreExtendArea(rows int, mp *mpool.MPool) error {
```

Comment:
Why don't you add area size as a parameter so that other types can also use it?

Reply:
PreExtendArea has same arguments as PreExtend
```go
func (v *Vector) PreExtend(rows int, mp *mpool.MPool) error {

```

And, sizes of other types are handled by this logic in PreExtendArea
```go
// get the size required for storing new rows
	var vSize int
	switch v.typ.Oid {
	case types.T_array_float32:
		vSize = 4 * int(v.typ.Width)
	case types.T_array_float64:
		vSize = 8 * int(v.typ.Width)
	default:
		vSize = v.typ.TypeSize()
	}
	vlen := vSize * rows
```